src/spartan/polynomial.rs

```diff
@@ -69,10 +100,13 @@ impl<Scalar: PrimeField> MultilinearPolynomial<Scalar> {
     self.Z.len()
   }
 
+  // NOTE: this is equivalent to Espresso/HyperPlonk's 'fix_last_variables' method
   pub fn bound_poly_var_top(&mut self, r: &Scalar) {
     let n = self.len() / 2;
 
     let (left, right) = self.Z.split_at_mut(n);
+
+    // XXX: What does this do? Seems like nothing?
```

Comment:
Good catch! I think in the past, we weren't resizing the Z vector, so this was necessary, but we do resize Z (in the code below), so this code is likely not needed (I'll double-check).